Sync repo templates ⚙

coreosbot-releng opened the pull request:
Created by [GitHub workflow](https://github.com/coreos/repo-templates/actions/workflows/sync.yml) ([source](https://github.com/coreos/repo-templates/blob/main/.github/workflows/sync.yml)).

Sync with coreos/repo-templates@1e012242aef09a4bfed0250bc4a06373df5c28c1.

travier commented:
```
error: hiding a lifetime that's elided elsewhere is confusing
   --> src/util/dhcp.rs:148:27
    |
148 |     fn active_connections(&self) -> zbus::Result<Vec<zvariant::ObjectPath>>;
    |                           ^^^^^                      -------------------- the same lifetime is hidden here
    |                           |
    |                           the lifetime is elided here
    |
    = help: the same lifetime is referred to in inconsistent ways, making the signature confusing
    = note: `-D mismatched-lifetime-syntaxes` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(mismatched_lifetime_syntaxes)]`
help: use `'_` for type paths
    |
148 |     fn active_connections(&self) -> zbus::Result<Vec<zvariant::ObjectPath<'_>>>;
    |                                                                          ++++

error: hiding a lifetime that's elided elsewhere is confusing
   --> src/util/dhcp.rs:157:21
    |
157 |     fn dhcp4_config(&self) -> zbus::Result<zvariant::ObjectPath>;
    |                     ^^^^^                  -------------------- the same lifetime is hidden here
    |                     |
    |                     the lifetime is elided here
    |
    = help: the same lifetime is referred to in inconsistent ways, making the signature confusing
help: use `'_` for type paths
    |
157 |     fn dhcp4_config(&self) -> zbus::Result<zvariant::ObjectPath<'_>>;
    |                                                                ++++

error: hiding a lifetime that's elided elsewhere is confusing
   --> src/util/dhcp.rs:166:16
    |
166 |     fn options(&self) -> Result<HashMap<String, zvariant::Value>>;
    |                ^^^^^                            --------------- the same lifetime is hidden here
    |                |
    |                the lifetime is elided here
    |
    = help: the same lifetime is referred to in inconsistent ways, making the signature confusing
help: use `'_` for type paths
    |
166 |     fn options(&self) -> Result<HashMap<String, zvariant::Value<'_>>>;
    |                                                                ++++
```

coreosbot committed:
Sync repo templates ⚙

Sync with coreos/repo-templates@e50202b0f0a498d5231972f4caaf8a0985f0b3cc.

travier committed:
util/dhcp: Fix clippy lints

Fix the followint lint:

```
warning: hiding a lifetime that's elided elsewhere is confusing
   --> src/util/dhcp.rs:148:27
    |
148 |     fn active_connections(&self) -> zbus::Result<Vec<zvariant::ObjectPath>>;
    |                           ^^^^^                      ^^^^^^^^^^^^^^^^^^^^ the same lifetime is hidden here
    |                           |
    |                           the lifetime is elided here
    |
    = help: the same lifetime is referred to in inconsistent ways, making the signature confusing
    = note: `#[warn(mismatched_lifetime_syntaxes)]` on by default
help: use `'_` for type paths
    |
148 |     fn active_connections(&self) -> zbus::Result<Vec<zvariant::ObjectPath<'_>>>;
    |
```
